### PR TITLE
Add two boards

### PR DESCRIPTION
**Pull request by Hyask**
Here are three more boards that we are using at Free Electrons:
  * Armada 8040 DB
  * OrangePi PC
  * Allwinner A23 EVB

### CODE REVIEW

File: lib/device_map.py [hunk 1]
```diff
@@ -171,6 +171,16 @@
                        'fastboot': False,
                        'mach': 'mvebu'}
 
+armada_8040_db = {'device_type': 'armada-8040-db',
+                       'templates': ['generic-arm64-dtb-kernel-ci-boot-template.json'],
+                       'kernel_defconfig_blacklist': [],
+                       'defconfig_blacklist': ['arm64-allmodconfig'],
+                       'kernel_blacklist': ['v4.6'],
```
Comment (QSchulz) [inline lib/device_map.py]: Hi Florent,

The Armada 8040 DB DTS has only been merged in 4.9 apparently so this kernel version blacklisting is useless.

Comment (Hyask) [reply lib/device_map.py]: Fixed! :)

**Comment by mattface** [comment]
Thanks for the patch.
Do these boards all boot upstream? And have a maintainer that will make fixes if they start to fail?

**Comment by khilman** [comment]
In particular, to they boot older (stable) kernels?

Also, are the LAVA device-types for these boards submitted to upstream LAVA?

**Comment by Hyask** [comment]
The tests were running when I left, and I have now a week in holidays, I'll tell you that when I'll come back (27 Feb)

The device-types are not yet submitted either to LAVA, but it will come! :)
Slap @QSchulz for not telling me this was necessary! :D

**Comment by khilman** [comment]
It's not the fault of @QSchulz either since we haven't really enforced that in the past, but we need to enforce that the LAVA devices are all upstream too.

**Comment by Hyask** [comment]
I see. I think I'll need to apologize to @QSchulz ... Maybe it'll cost me a beer ;)

Anyway, I'll make the effort to upstream the device-types to LAVA, and I'll tell you when it's done.

**Comment by Hyask** [comment]
OrangePi PC passed all the stable and mainline tests, as told in the README.

Sadly, I'm still waiting for the two other boards to be put back online before running the tests on them.

**Comment by Hyask** [comment]
Okay, here is the status:
OrangePi PC and Allwinner A23 EVB passed all the jobs smoothly, except the few jobs named `mainline-v4.10-rc6-188-g412e6d3fec24-*`.   
On this commit only they have failed. I think it's not the fault of the configuration, but rather of this particular commit.

On the side of the Armada 8040 DB, it has for now failed all the jobs, but since it had had a bootloader update between when I added it and when I run the jobs, it may be related. Anyway, I'm just going to remove it from the pull request, and we'll take time later to debug.

Concerning upstreaming the `device-type`, is it here you want me to put them: https://git.linaro.org/lava/lava-dispatcher.git/tree/lava_dispatcher/default-config/lava-dispatcher/device-types ?

**Comment by khilman** [comment]
Yes, that is the right place for the device types for V1 devices.

The LAVA team has been rather slow in the past in reviewing/merging new device types however, so hopefully someone in Linaro can help accelerate that process a bit.

**Comment by Hyask** [comment]
Okay, I had the patch ready, so I'll try to send them tomorrow.

**Comment by khilman** [comment]
Note that sending the patch to the list may not be enough.  I think they prefer the gerrit reviews.

**Comment by Hyask** [comment]
Yep, I've seen that in their doc, but I haven't been able to find out how to send something to Gerrit, I'll ask them tomorrow on IRC.

**Comment by Hyask** [comment]
Okay, the patchs are in review, I hope they'll soon be merged

**Comment by QSchulz** [comment]
@Hyask you actually need to assign a guy at Linaro to review your code (or a default one that will assign a reviewer instead of himself).

> If a review in gerrit only has the submitter and lava-bot in the
reviewers list, it will not get picked up by the LAVA software team.
(The lack of reviewers is, as a convention, interpreted as a review
which is a work-in-progress and is not ready for being merged.) These
reviews will show up in the "My Changes" view in gerrit when you login
but will not show up to anyone else. I hope this explains why these
reviews have had no comments from the team yet.

(mail from Neil Williams - head developer - to myself).

Anyway, we don't actually need the patches to be merged to merge this PR. It's just a way to be sure all labs can have the same configuration files and ease the integration of said boards in other labs.

@khilman I see the point of sending patches for newly added boards as a reference for people who want to add such boards in their labs but there is no point anymore to do that for LAVA v1 configuration files as it is deprecated and is planned to be entirely removed in a few months.

**Comment by khilman** [comment]
@QSchulz as much as the LAVA developers think that v1 will be removed, they are still a long ways from having V2 functionality match V1 functionality.  

Therefore, for kernelci.org, we want V1 device-types merged into mainline LAVA so they can be shared across multiple labs.  The other benefit of having them merged is that others can help migrate them to V2.

**Comment by Hyask** [comment]
The patches are merged in LAVA

**Pull request merged**